### PR TITLE
Add os-specific telemetry metadata.

### PR DESCRIPTION
**Pull request by josecv**
Signed-off-by: Jose Cortes <josecortes@datawire.io>

## Description

A few sentences describing the overall goals of the pull request's commits.

## Checklist

 - [ ] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to either submit a docs PR, or tell Matt about the necessary documentation changes.
 - [x] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

### CODE REVIEW

**Comment by josecv** [comment]
@maxlegault i've added you so you can see the new metrics/let us know if you have any feedback on them

File: pkg/client/scout/os_metadata_linux.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+func getOsMetadata(ctx context.Context) map[string]interface{} {
+	osMeta := map[string]interface{}{}
+	osMeta["os_docker"] = isDocker(ctx)
+	osMeta["os_wsl"] = isWSL(ctx)
+	f, err := os.Open("/etc/os-release")
+	if err != nil {
+		dlog.Warnf(ctx, "Unable to open /etc/os-release: %v", err)
+		return osMeta
+	}
```
Comment (LukeShu) [inline pkg/client/scout/os_metadata_linux.go:43]: If `/etc/os-release` does not exist, you should fall back to `/usr/lib/os-release`.  See `man os-release` for more information.

File: pkg/client/scout/os_metadata_linux.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/datawire/dlib/dlog"
+)
+
+func isDocker(ctx context.Context) bool {
+	cgroups, err := ioutil.ReadFile("/proc/1/cgroup")
+	if err != nil {
+		dlog.Warnf(ctx, "Unable to read /proc/1/cgroup: %v", err)
+		return false
+	}
+	return strings.Contains(string(cgroups), "/docker/")
```
Comment (LukeShu) [inline pkg/client/scout/os_metadata_linux.go:19]: FWIW, `systemd-detect-virt` and friends (which I generally trust to be good generic utilities that people are mislead in to thinking are only useful with systemd), check for Docker by checking for the `/.dockerenv` file. https://github.com/systemd/systemd/blob/main/src/basic/virt.c#L597

I'd love if we could just include the `systemd-detect-virt` output, but we can't rely on it being there :(

**Review by LukeShu — APPROVED** [review]
I'm going ahead and clicking "approve", contingent on falling back to `/usr/lib/os-release`

**Pull request merged**